docs/core/glt.md

```diff
@@ -0,0 +1,127 @@
+# Using GLT for unstructured logs
+
+GLT is a version of CLP-unstructured that specialize in search performance, at cost of 
+higher compression memory usage. 
+During compression, the log types are grouped together into tables that can be compressed and searched more efficiently. 
+In general, GLT has higher compression ratio than CLP, but it could result in slightly worse compression ratio depending
+on the nature of input logs. 
+
+For unstructured (plain text) logs, you can compress, decompress, and search them using the `glt` binary 
+described below.
+
+## Contents
+
+* [Compression](#compression)
+* [Decompression](#decompression)
+* [Search](#search)
+* [Utilities](#utilities)
+  * [`make-dictionaries-readable`](#make-dictionaries-readable)
+* [Current limitations](#current-limitations)
+
+## Compression
+
+Usage:
+
+```shell
+./glt c [<options>] <archives-dir> <input-path> [<input-path> ...]
+```
+
+* `archives-dir` is the directory that archives should be written to.
+* `input-path` is any new-line-delimited JSON (ndjson) log file or directory containing such files.
+* `options` allow you to specify things like a custom percentage threshold for combined logtype tables
+    * For a complete list, run `./glt c --help`
+
+### Examples
+
+**Compress `/mnt/logs/log1.log` and output archives to `/mnt/data/archives1`:**
+
+```shell
+./glt c /mnt/data/archives1 /mnt/logs/log1.log
+```
+
+## Decompression
+
+Usage:
+
+```bash
+./glt x <archives-dir> <output-dir>
+```
+
+* `archives-dir` is a directory containing archives.
+* `output-dir` is the directory that decompressed logs should be written to.
+
+### Examples
+
+**Decompress all logs from `/mnt/data/archives1` into `/mnt/data/archives1-decomp`:**
+
+```bash
+./glt x /mnt/data/archives1 /mnt/data/archives1-decomp
+```
+
+## Search
+
+Usage:
+
+```shell
+./glt s [<options>] <archives-dir> <wildcard-query> [<file-path>]
+```
+
+* `archives-dir` is a directory containing archives.
+* `wildcard-query` is a wildcard query where:
+    * the `*` wildcard matches 0 or more characters;
+    * the `?` wildcard matches any single character.
+* `options` allow you to specify things like a time-range filter.
+    * For a complete list, run `./glt s --help`
+
+> [!TIP]
+> Adding spaces (when possible) at the begin and the end of the wildcard-query can improve GLT's search performance,
+> as GLT doesn't need to consider implicit wildcards during query processing.
```

Comment:
```suggestion
> Adding spaces (when possible) at the beginning and the end of the wildcard-query can improve GLT's
> search performance, since GLT won't need to consider implicit wildcards during query processing.
```